### DISCUSSION TITLE
Unable to remove a secret file from my repo

### DISCUSSION THREAD

**Discussion by realhackcraft** [community/community#38607] [Other Feature Feedback, Questions, & Ideas] [answered]
I accidentally committed a .env file in a private repo. But now, I want to make the repo public so I want to remove the .env file from the history. I ran ` git filter-repo --invert-paths --path .env` and `git push origin --force --all` so now my local branch has no record of that file, but GitHub still has that file cached. 

What am I doing wrong?

**Comment by realhackcraft** [comment]
![Screen Shot 2022-11-08 at 5 39 04 PM](https://user-images.githubusercontent.com/65378120/200691373-15e70d2f-db07-4660-a750-df2be34ee7d8.png)
![Screen Shot 2022-11-08 at 5 39 37 PM](https://user-images.githubusercontent.com/65378120/200691455-1f1db2d7-df77-4bf4-81a6-d3e4de722ad7.png)
Don't worry, I changed the token but still want to remove that file entirely

**Comment by airtower-luna** [comment, marked as answer]
Is that commit on any branch on GitHub, or just visible by its commit ID?

If it's on a branch, you didn't clean the branch for some reason, maybe you didn't have a local copy of it when you did the cleaning?

Either way the commit will still be accessible by its ID. As the red warning box in the documentation on [removing sensitive data](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository) says unreachable commits usually won't be cleaned up, you need to contact support if you want them really gone.

**Reply by realhackcraft** [reply]
I'm pretty sure its still visible on the branch.
If I contact support, will they also remove it from the branch?
Also, where can I contact support?

**Reply by realhackcraft** [reply]
The only way I know to get to the file is to go to the history that says "x commits" with a clock on the left, and click on the commit in which I added that file.

**Reply by airtower-luna** [reply]
> If I contact support, will they also remove it from the branch?

No, you need to actually clean up the branch first, otherwise the unmodified commit will still be needed to keep the repository intact.

And yes, what you describe means the commit is still reachable. I can't say exactly how to fix it without looking at your repository, but [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) might help. Another thing that might simplify things: Make sure you have as few branches as possible in your GitHub repository. That will make it much easier to check them all.

**Reply by realhackcraft** [reply]
Well, I changed the local history, but can't seem to change the github one. Can you tell me which command to use?

**Reply by airtower-luna** [reply]
In principle, push the branches you changed with [`git push --force`](https://git-scm.com/docs/git-push) (add your remote and branch). Double check the branches are correct, because there's no way to recover if you push the wrong thing unless you have a backup. :warning:

**Reply by realhackcraft** [reply]
yep i did that. in the webstorm ide's git interface, there is no .env file. but on GitHub, you can still see it by going to 
https://github.com/realhackcraft/my-repo/blob/0b67b5c304223bbefaa1c23c4c0614e0bec61d8d/.env

**Reply by realhackcraft** [reply]
i asked github support. do you know how long do they normally take to remove such file?

**Reply by airtower-luna** [reply]
The repository seems to be private, so I can't check either way. :sweat_smile: 

And no, I don't know. From what I've seen around here response times can vary widely.

**Reply by realhackcraft** [reply]
ok thanks!
